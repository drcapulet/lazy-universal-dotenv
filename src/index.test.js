import { getEnvironment } from "."

test("Sets up process.env", () => {
  expect(process.env.APP_TEST_PATH).toBe("one/two/three.js")
  expect(process.env.TEST_IGNORE).toBe("123")
})

test("Ignores non app-specific settings", () => {
  const { raw, stringified, webpack } = getEnvironment()
  expect(raw).toMatchSnapshot()
  expect(stringified).toMatchSnapshot()
  expect(webpack).toMatchSnapshot()
})

test("Supports manually defined envs", () => {
  process.env.APP_DATA = "yellow"
  const { raw, stringified, webpack } = getEnvironment()
  expect(raw).toMatchSnapshot()
  expect(stringified).toMatchSnapshot()
  expect(webpack).toMatchSnapshot()
  delete process.env.APP_DATA
})
