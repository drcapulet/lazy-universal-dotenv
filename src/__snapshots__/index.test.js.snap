// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ignores non app-specific settings 1`] = `
Object {
  "APP_TEST_OVERRIDE": "overridden",
  "APP_TEST_PATH": "one/two/three.js",
  "NODE_ENV": "test",
}
`;

exports[`Ignores non app-specific settings 2`] = `
Object {
  "APP_TEST_OVERRIDE": "\\"overridden\\"",
  "APP_TEST_PATH": "\\"one/two/three.js\\"",
  "NODE_ENV": "\\"test\\"",
}
`;

exports[`Ignores non app-specific settings 3`] = `
Object {
  "process.env": Object {
    "APP_TEST_OVERRIDE": "\\"overridden\\"",
    "APP_TEST_PATH": "\\"one/two/three.js\\"",
    "NODE_ENV": "\\"test\\"",
  },
}
`;

exports[`Supports manually defined envs 1`] = `
Object {
  "APP_DATA": "yellow",
  "APP_TEST_OVERRIDE": "overridden",
  "APP_TEST_PATH": "one/two/three.js",
  "NODE_ENV": "test",
}
`;

exports[`Supports manually defined envs 2`] = `
Object {
  "APP_DATA": "\\"yellow\\"",
  "APP_TEST_OVERRIDE": "\\"overridden\\"",
  "APP_TEST_PATH": "\\"one/two/three.js\\"",
  "NODE_ENV": "\\"test\\"",
}
`;

exports[`Supports manually defined envs 3`] = `
Object {
  "process.env": Object {
    "APP_DATA": "\\"yellow\\"",
    "APP_TEST_OVERRIDE": "\\"overridden\\"",
    "APP_TEST_PATH": "\\"one/two/three.js\\"",
    "NODE_ENV": "\\"test\\"",
  },
}
`;
